ci/azure-pipelines.yml

```diff
@@ -125,12 +125,11 @@ stages:
 
         strategy:
           matrix:
-            ubuntu22-mainline:
+            ubuntu2310:
+              BOX_NAME: ubuntu2310
+            ubuntu2204:
               BOX_NAME: ubuntu2204
-              KERNEL: mainline
```

Comment:
if think we should remove gcc-related building function in script but otherwise keep the provision1/provision2 for when we need kernel >6.5.

Reply:
Removed the GCC install, build passed.